### PR TITLE
Graphite table support

### PR DESCRIPTION
**Pull request by mhosken**
This PR adds support for the Graphite font tables: Feat, Glat, Gloc, Silf, Sill providing ttx XML support as well.

### CODE REVIEW

**Comment by brawer** [comment]
Nice! Do you plan adding unit tests? Something like [this](https://github.com/fonttools/fonttools/blob/master/Tests/ttLib/tables/_l_c_a_r_test.py), for example. Tests help a lot when maintaining a large codebase.

**Comment by anthrotype** [comment]
Thank you Martin for working on this! 👍 
Yes, some simple tests would be nice to have.

one minor suggestion I was thinking of, that `fontTools/ttLib/tables/grUtils.py` module could named more explicitly `graphiteUtils` maybe?

File: Lib/fontTools/ttLib/tables/G__l_o_c.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+from fontTools.misc import sstruct
+from fontTools.misc.textTools import safeEval
+from itertools import *
+from functools import partial
+from . import DefaultTable
+import array
+import struct
+import operator
+import warnings
+from _ast import Num
```
Comment (anthrotype) [inline Lib/fontTools/ttLib/tables/G__l_o_c.py]: this import seems unused

**Comment by behdad** [comment]
Are the specs for these tables available online?
Can these tables be implemented using the otData machinery?  I prefer not adding more hand-coded table parsing if we can avoid it.

**Comment by mhosken** [comment]
The specs are available as part of the Graphite source tree in doc/GTF.txt. I did consider trying to use the otData machinery but it would have involved so many extensions to the core supporting types and a huge growth in what are already big files that I felt it better to use the old approach. I did try to use it for the Feat table, but gave up after it refused to accept any table with a version > 1! The otData machinery is very tightly tied together and hard to open up and as such is really only good for tables that are part of the OT machinery and that can make best use of the monolithic structure.

Now, if I were starting again on the Graphite binary format, I would probably go for something much closer in style to the OT structures to make use of such machinery. But even then I don't know how close I would get.

File: Lib/fontTools/ttLib/tables/G__l_o_c.py (hunk 1 above)
Comment (mhosken) [reply Lib/fontTools/ttLib/tables/G__l_o_c.py]: Cleared out the redundant imports there. I'll use graphiteUtils if you use opentypeUtils :) gr || ot just as graphite || opentype.

**Comment by mhosken** [comment]
Test are in.

**Comment by mhosken** [comment]
I'm done fiddling subject to fixing up review comments.

**Pull request merged**